mat/matrix_test.go

```diff
@@ -782,9 +740,9 @@ func TestMulVecToer(t *testing.T) {
 						}
 						x = dst
 					case 2:
-						x = &rawVector{basicVector{random(n)}}
+						x = &rawVector{(*basicVector)(NewVecDense(n, random(n)))}
```

Comment:
Why not using `asBasicVector` here?

Reply:
Because `asBasicVector` returns a `Vector` but `rawVector` wraps `*basicVector` so I would need to extract the concrete type anyway. Maybe the `asBasicSomething` functions should return concrete types instead of interfaces. Regarding `rawVector`, I'd like to revisit this test and look if I can remove this type and simplify the test.